Document block reward contract

andresilva opened the pull request.

andresilva commented:
@Tbaut Thanks, updated.

Tbaut commented on Block-Reward.md:
```diff
@@ -0,0 +1,65 @@
+### AuthorityRound
+
+AuRa is currently the only engine supporting a block reward contract. It can be enabled by defining
+the address of the block reward contract in the chainspec and a block to activate it:
```
Can we link chain spec (/Chain-specification) and add these params in https://github.com/paritytech/wiki/blob/master/Chain-specification.md ?

Tbaut commented:
Thanks @andresilva. I added a couple others things. Should we address [Afri's comment](https://github.com/paritytech/parity/pull/8419#issuecomment-389303455) in a dedicated "limitations" section? that'd be great :)

Tbaut commented on Block-Reward.md:
```diff
@@ -0,0 +1,65 @@
+## Block Reward
```
Please use front matter instead of a title:
```
---
title: Block Reward
---
```
and set any other title as an h2, therefore with `##` instead of  `###`

5chdn replied:
Can we name this site Block Reward Contract? Because it does not cover other means to specify the block reward.

5chdn commented on _includes/_Sidebar.md:
```diff
@@ -68,6 +68,7 @@
     - [Aura](Aura)
   - [Validator Set](Validator-Set)
     - [Validator contracts](Validator-contracts)
+  - [Block Reward](Block-Reward)
```
Isn't that a sub-site of AuRa?

Tbaut commented:
ping @andresilva :)

andresilva commented:
@Tbaut Sorry for taking so long I forgot about this PR. I addressed all comments except moving `Block Reward Contract` to a subsection of Aura. Let me know what you think.
